src/lib/cspauth/csp_auth.go

```diff
@@ -0,0 +1,119 @@
+package cspauth
+
+import (
+	"context"
+	"fmt"
+	"github.com/vmware-tanzu/cloud-native-security-inspector/src/lib/log"
+	"github.com/vmware-tanzu/cloud-native-security-inspector/src/lib/retry"
+	v12 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"math"
+	"time"
+)
+
+const (
+	tokenMaxAgeSeconds     = 1700
+	apiToken               = "API_TOKEN"
+	cspConfigMapName       = "csp-config"
+	governorTokenExpiresIn = "governorAccessTokenExpiresIn"
+	governorAccessTokenKey = "governorAccessToken"
+	Retry                  = 3
+)
+
+var RetryDelay time.Duration = 5
+
+// Provider is an interface to interact with an authorization service
+type Provider interface {
+	// GetBearerToken retrieves a short-lived access token to use in a single HTTP request
+	GetBearerToken(kubernetes.Interface, context.Context, string, string) (string, error)
+}
+
+type CspAuth struct {
+	CspClient CSPClient
+
+	apiToken string
+}
+
+func (a *CspAuth) GetBearerToken(clientSet kubernetes.Interface, ctx context.Context, cspSecretNamespace string, cspSecretName string) (string, error) {
+	configMap, err := getOrCreateCSPConfigMap(clientSet, ctx, cspSecretNamespace)
```

Comment:
IMO using a configmap as "cache" is not the right approach. Originally configmaps are not intended for that kind of usage. You always could use a volume for that, tho.

However, keep in mind that the access token is sensitive information so, in case having to choose one of them to act as acache (which I'm not fan of), we would have to use a secret.

Reply:
Done.